embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/EmbeddedIndexTaskTest.java

```diff
@@ -102,7 +102,12 @@ public void test_runIndexTask_forInlineDatasource()
       start = start.plusDays(1);
     }
 
-    // Wait for segments to be queryable
+    // Wait for all segments to be loaded and queryable
```

Comment:
I tried fixing this in #18196 with a different approach, where there's a hook into the Broker to look at its server view. I think it will be more robust, because it ensures the Broker actually sees all segments (in theory, with this inspection of Coordinator metrics, they could be loaded on Historicals but the Broker might not know about them yet).

Reply:
@gianm , should we proceed with this for now until we add the new Broker monitor for emitting segment count per datasource?

Reply:
Sounds good, thanks!